docs/content/concepts/assets/graph-backed-assets.mdx

```diff
@@ -0,0 +1,208 @@
+
+In the example below, Dagster creates an asset with key `middle_asset` from the `middle_asset` graph. Just like assets defined via <PyObject object="asset" decorator />, each argument to the decorated graph function is an upstream asset name. `middle_asset` depends on `upstream_asset`, and `downstream_asset` depends on `middle_asset`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_basic_dependencies endbefore=end_basic_dependencies
+from dagster import AssetsDefinition, asset, graph
+
+
+@asset
+def upstream_asset():
+    return 1
+
+
+@graph
+def middle_asset(upstream_asset):
+    return add_one(upstream_asset)
+
+
+middle_asset = AssetsDefinition.from_graph(middle_asset)
+
+
+@asset
+def downstream_asset(middle_asset):
+    return middle_asset + 1
+```
+
+When your graph returns multiple outputs, Dagster infers each output name to be the outputted asset key. In the below example, `two_assets_graph` accepts `upstream_asset` and outputs two assets, `first_asset` and `second_asset`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_basic_dependencies_2 endbefore=end_basic_dependencies_2
+from dagster import AssetsDefinition, GraphOut, graph
+
+
+@graph(out={"first_asset": GraphOut(), "second_asset": GraphOut()})
+def two_assets_graph(upstream_asset):
+    one, two = two_outputs(upstream_asset)
+    return {"first_asset": one, "second_asset": two}
+
+
+two_assets = AssetsDefinition.from_graph(two_assets_graph)
+```
+
+### Defining explicit dependencies for graph-backed assets
+
+You can also define dependencies for graph-backed assets explicitly via the `asset_keys_by_input_name` and `asset_keys_by_output_name` arguments to `from_graph`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_explicit_dependencies endbefore=end_explicit_dependencies
+from dagster import AssetsDefinition, GraphOut, graph
+
+
+@graph(out={"one": GraphOut(), "two": GraphOut()})
+def return_one_and_two(zero):
+    one, two = two_outputs(zero)
+    return {"one": one, "two": two}
+
+
+explicit_deps_asset = AssetsDefinition.from_graph(
+    return_one_and_two,
+    keys_by_input_name={"zero": AssetKey("upstream_asset")},
+    keys_by_output_name={
+        "one": AssetKey("asset_one"),
+        "two": AssetKey("asset_two"),
+    },
+)
+```
+
+### Advanced: Subsetting graph-backed assets
+
+By default, when executing a graph-backed asset, every asset produced by the graph must be materialized. This means that attempting to selectively execute a subset of assets defined in the graph-backed asset will result in an error.
+
+If the underlying computation is sufficiently flexible to selectively output a subset of assets, a graph-backed asset can be subsetted. For example, let’s say we wanted to define a graph-backed asset with the structure depicted in the image below. In this case, we want to independently materialize `foo_asset` and `baz_asset`.
+
+<Image
```

Comment:
I've been working on making our diagram styles consistent ([example here](https://docs.dagster.io/concepts/io-management/io-managers)) - I'd be happy to make a version of this diagram that fits that style.

Reply:
That sounds great, thanks so much! I can add that in once you make it

Reply:
I'll knock it out later today!